### PR TITLE
refactor to be usable and testable without modal

### LINKED ISSUE #1: implement 1H NMR endpoint

### LINKED ISSUE #2: implement periodic table endpoint

### PR DESCRIPTION
**Pull request by kjappelbaum**
This is a major change to the codebase. In some parts, it leads to more complications, but in other parts, it makes things a lot easier. 

It would be great if the tools/functions here could be used and tested independently from the modal (to increase the potential for reusability—also for some of our projects). Hence, I refactored the project into an installable Python package. 

I'd now define all the tools as Python functions that can be used and tested as conventional Python functions. In addition, they will now be imported into the `modal.py` module with the relevant dependencies. 
Right now, the modal tests still test some functionality. However, they likely should be refactored to only test if the call/the promised signature works. 

A great advantage of modal is that we do not need to worry about dependency conflicts. However, we can still have that if needed. 

For clarity, I also started renaming all methods `get_x`. We might find a better convention later, but I wanted to unify what we have. 

I also have the RXN lookup (#6) built as FAISS DB (https://kjappelbaum--rxn-search-webapp-search.modal.run/doc), but I'll leave this for a separate PR once this here is merged as it will be too messy otherwise

### CODE REVIEW

**Comment by kjappelbaum** [comment]
The modal tests work for me locally, perhaps there is some issue with the secret on GitHub.

**Comment by kjappelbaum** [comment]
Oh ... I forgot that secrets in PRs are more complex https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: PR Actions
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled]
```
Comment (kjappelbaum) [inline .github/workflows/pr.yml:4]:
```suggestion
    types: [opened, reopened, synchronize]
```

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: PR Actions
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
```
Comment (kjappelbaum) [inline .github/workflows/pr.yml]:
```suggestion
    types: [opened, reopened]
```

**Comment by kjappelbaum** [comment]
ok - I should have read the docs more carefully. The `pull_request_target` action is only triggered for *internal* PRs and not for PRs from forks.

I do not see a good way to provide secrets to actions on PRs from forks. Also the dispatch does not run on PRs. 

In this way, the refactoring at least gives us some testability.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -12,22 +12,27 @@ jobs:
...
-          cache: pip
-      - name: Install Modal
-        run: pip install -r requirements.txt
+      - name: Install Poetry
+        run: pip install poetry
+      - name: Install dependencies
+        run: poetry install
       - name: Deploy to dev
-        run: modal deploy tools/deploy.py
+        run: poetry run chemenv deploy
```
Comment (whitead) [inline .github/workflows/deploy.yml]: Can we try to use `uv` instead?

Comment (whitead) [reply .github/workflows/deploy.yml]: Optional - just we've found it to speed up build times quite a bit

**Review by whitead — APPROVED** [review]
I agree with this idea - thanks for taking a crack at it!

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (kjappelbaum) [reply .github/workflows/deploy.yml]: sure, I'm also moving ChemBench to `uv` atm

**Comment by kjappelbaum** [comment]
Thanks for giving it a look, @whitead!

**Pull request merged**